components/provisioner/internal/installation/download/components.go

```diff
@@ -0,0 +1,91 @@
+package download
```

Comment:
I would consider changing package structure a bit. I would suggest to group components related to parallel-install and Kyma Operator in a separate folder structure.